deploy/scripts/aad-register.ps1

```diff
@@ -84,7 +84,26 @@ Function Select-Context() {
         return $context
     }
     catch {
-        throw "The login to the Azure account was not successful."
+        try {
+            # for the case where prompting web browser is not applicable, we will prompt for DeviceAuthentication login.
+            $azProfile = Connect-AzAccount `
+            -UseDeviceAuthentication `
+            -TenantId $script:tenantId `
```

Comment:
`-TenantId $script:tenantId` part might already be included by `@tenantIdArg` parameters below. It also should do the check for empty `$script:tenantId`. Please have a look.

Reply:
Yes, I agree, thanks for pointing out. I have edited again.